app/scripts/contentscript.js

```diff
@@ -522,6 +527,38 @@ function logStreamDisconnectWarning(remoteLabel, error) {
   );
 }
 
+/**
+ * The function notifies inpage when the extension stream connection is ready. When the
+ * 'metamask_chainChanged' method is received from the extension, it implies that the
+ * background state is completely initialized and it is ready to process method calls.
+ * This is used as a notification to replay any pending messages in MV3.
+ *
+ * @param {object} msg - instance of message received
+ */
+function extensionStreamMessageListener(msg) {
+  if (
+    METAMASK_EXTENSION_CONNECT_SENT &&
```

Comment:
since we are adding the `extensionStreamMessageListener` listener on each `setupExtensionStreams` call, could we remove this flag and add logic to remove the `extensionStreamMessageListener` listener after the message has been sent?

Reply:
I tried to remove this listener `extensionStream.on('data', extensionStreamMessageListener);` but it does not work. There is an `off` method in api, but it does not works and rather throw error that `off` method does not exist.